internal/client/client.go

```diff
@@ -0,0 +1,232 @@
+// Package client contains functions for communicating with the API server
+// Copyright (C) 2020 by Red Hat, Inc.
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"time"
+)
+
+// Request handles sending the request, handling errors, returning the response
+// socket is the path to a Unix Domain socket
+// path is the full URL path, including query strings
+// body is the data to send with POST
+// headers is a map of header:value to add to the request
+//
+// If it is successful a http.Response will be returned. If there is an error, the response will be
+// nil and error will be returned.
+func Request(socket, method, path, body string, headers map[string]string) (*http.Response, error) {
+	client := http.Client{
```

Comment:
I think this should be passed in from the highest level, making it more flexible and type-safe.

Reply:
I think I need more context here. Pass what in?

Reply:
The `http.Client{}` object. Rather than the `socket` as a string.

Reply:
Ah! right. That would also allow for the timeout to be adjusted at setup time.